myriad-scheduler/src/main/java/org/apache/myriad/scheduler/event/handlers/StatusUpdateEventHandler.java

```diff
@@ -75,21 +90,37 @@ public void onEvent(StatusUpdateEvent event, long sequence, boolean endOfBatch)
         schedulerState.removeTask(taskId);
         break;
       case TASK_FAILED:
-        // Add to pending tasks
         offerLifecycleManager.declineOutstandingOffers(task.getHostname());
-        schedulerState.makeTaskPending(taskId);
+        // Add to pending tasks if the failed task was not actually a killed task
+        if (taskIsKillable(taskId)) {
+          schedulerState.removeTask(taskId);  
+          LOGGER.info("Removed killable, failed task with id {}", taskId);
+        } else {
+          schedulerState.makeTaskPending(taskId);          
+        }  
```

Comment:
All requested changes done and pushed to git

On Wed, Jul 13, 2016 at 9:38 AM, John Yost <soozandjohnyost@gmail.com>
wrote:

> Ah, good catch, will do
>
> On Wed, Jul 13, 2016 at 8:53 AM, Darin <notifications@github.com> wrote:
>
>> In
>> myriad-scheduler/src/main/java/org/apache/myriad/scheduler/event/handlers/StatusUpdateEventHandler.java
>> <https://github.com/apache/incubator-myriad/pull/84#discussion_r70618603>
>> :
>>
>> >          offerLifecycleManager.declineOutstandingOffers(task.getHostname());
>> > -        schedulerState.makeTaskPending(taskId);
>> > +        // Add to pending tasks if the failed task was not actually a killed task
>> > +        if (taskIsKillable(taskId)) {
>> > +          schedulerState.removeTask(taskId);
>> > +          LOGGER.info("Removed killable, failed task with id {}", taskId);
>> > +        } else {
>> > +          schedulerState.makeTaskPending(taskId);
>> > +        }
>>
>> this if/else block is replicated below modulo a log statement. Private
>> method?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/apache/incubator-myriad/pull/84/files/b4fbad0582bb71b35968b1cdfb4e85e629ec3758#r70618603>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AKSRgT_6PLy96gnkn6uiVmP-7jQ_w81bks5qVN-sgaJpZM4JLUAy>
>> .
>>
>
>